File: backend/src/main/java/de/nak/iaa/sundenbock/dto/userDTO/UserDTO.java
package de.nak.iaa.sundenbock.dto.userDTO;

/**
 * Data Transfer Object (DTO) representing a basic user.
 * <p>
 * Contains the user's unique identifier and username.
 * </p>
 */
public record UserDTO(
        Long id,
        String username
) { }

File: frontend/src/app/features/users/data/user.dto.ts
/**
 * @file user.dto.ts
 *
 * Definiert die Data Transfer Object (DTO) Interfaces für Benutzer.
 * `UserDto` ist die Basisstruktur von der API.
 * `UserDetailDto` ist eine erweiterte Struktur, die zusätzlich
 * Rollen und Berechtigungen für einen Benutzer enthält.
 */
export interface UserDto {
  id: number;
  username: string;
  email: string;
  firstName: string;
  lastName: string;
  enabled: boolean;
  createdAt: string;
  updatedAt: string;
}

/**
 * Erweiterte Detailstruktur für GET /api/v1/users/{username}/details
 * enthält Rollen und Berechtigungen
 */
export interface UserDetailDto extends UserDto {
  roles: string[];
  permissions: string[];
}